File: projects/springstuff/part-02/src/main/java/workshop/spring/beans/HelloWorldConfig.java
package workshop.spring.beans;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class HelloWorldConfig {

    @Bean
    public KillerService killerService() {
        return new HelloWorldService();
    }

    @Bean
    public KillerApplication killerApplication(KillerService killerService) {
        return new KillerApplicationImpl(killerService);
    }
}

File: kubernetes/k8s_001/oxygen/src/main/kotlin/de/roskenet/oxygen/helloworld/HelloWorldConfig.kt
package de.roskenet.oxygen.helloworld

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class HelloWorldConfig {

    @Value("\${hello.greeting}")
    lateinit var greeting: String

    @Bean
    fun oxygenApi(): OxygenApi {
        return HelloWorld(greeting)
    }
}